packages/core/src/amazonq/extApi.ts

```diff
@@ -0,0 +1,31 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import vscode from 'vscode'
+import { VSCODE_EXTENSION_ID } from '../shared/utilities'
+import { ToolkitError } from '../shared'
+import { GenerateAssistantResponseCommandOutput, GenerateAssistantResponseRequest } from '@amzn/codewhisperer-streaming'
+import { FeatureAuthState } from '../codewhisperer/util/authUtil'
+
+export interface qAPI {
+    chatApi: {
+        chat(request: GenerateAssistantResponseRequest): Promise<GenerateAssistantResponseCommandOutput>
+    }
+    authApi: {
+        reauthIfNeeded(): Promise<void>
+        getChatAuthState(): Promise<FeatureAuthState>
+    }
+}
+
+/**
+ * Get the extension API for Q
+ */
+export async function getQAPI(): Promise<qAPI> {
```

Comment:
"amazonq" is the name for Amazon Q in the codebase. Acronyms are not case differently that other words ("HTML" => "Html", etc.).

```suggestion
export async function getAmazonqApi(): Promise<amazonqApi> {
```